Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -274,7 +275,32 @@ private static bool IsIonNamedProperty(PropertyInfo property)
         {
             return property.GetCustomAttribute(typeof(IonPropertyName)) != null;
         }
-        
+
+        /// <summary>
+        /// We only serde public, internal, and protected internal properties.
+        /// </summary>
+        private static bool HasValidAccessModifier(PropertyInfo propertyInfo)
+        {
+            var methodInfo = propertyInfo.GetGetMethod(true);
+
+            if (methodInfo is null)
+            {
+                return false;
+            }
+            else
+            {
+                return methodInfo.IsPublic || methodInfo.IsAssembly || methodInfo.IsFamilyOrAssembly;
+            }
+        }
+
+        /// <summary>
+        /// We only serde public, internal, and protected internal methods.
+        /// </summary>
+        private static bool HasValidAccessModifier(MethodInfo methodInfo)
```

Comment:
Invalid access modifier?

This check only checks properties that have public/internal/protected internal access modifiers. Otherwise it will return false.

Reply:
I'm curious for an example of something that would fail this check. If everything is either public, internal, or protected, then everything would pass this function and there'd be no need for it.

Reply:
We get all properties for the object type and then we filter using this function.  For example private properties are excluded (or fail this check).